code/headers/uarm_swift_pro.hpp

```diff
@@ -0,0 +1,86 @@
+#pragma once
+
+#include <cstring>
+#include <hardware_usart.hpp>
+#include <hwlib.hpp>
+#include <robot_arm_interface.hpp>
+#include <uarm_gcode_generator.hpp>
+#include <vector3.hpp>
+
+/**
+ * Class uArm_swift_pro provides a implementation for uArm swift pro.
+ */
+namespace r2d2::robot_arm {
+    class uarm_swift_pro_c : public robot_arm_interface_c {
+    protected:
+        const uint16_t default_speed = 1000;
+
+    private:
+        uarm_gcode_generator_c<50> gcode_generator;
+        hardware_usart_c usart_bus;
+
+    public:
+        /**
+         * Robot arm constructor, needs an usart bus for communication
+         */
+        uarm_swift_pro_c(const r2d2::hardware_usart_c &usart_bus);
+        /**
+         * robot arm constructor, creates a usart bus from the bautrate and port
+         **/
+        uarm_swift_pro_c(unsigned int &bautrate,
+                         r2d2::uart_ports_c &usart_port);
+
+        /**
+         * @brief Initialization function
+         * This function is for actions that need to be executed at
+         * initialization of the object
+         *
+         */
+        void init();
+        /**
+         * @brief Check connection
+         * Function to check whether the connection with the arm is right or not
+         *
+         * @return true means connection is right
+         * @return false means connection is wrong
+         */
+        bool check_connection();
+        /**
+         * @brief Sending command to arm
+         * Sends false as default
+         *
+         * @param command
+         * @return false when something went wrong with sending
+         * @return true when command correctly sent
+         */
+        bool send_command(const char *command);
+        /**
+         * Move specific joint of the arm
+         * Moves the robot arm joint [joint_id] in a given angle
+         *
+         */
+        void move_joint(const int &joint_id, const int &angle);
```

Comment:
Is this angle in radians or degrees?

Reply:
I am unsure of this since I have not made this function, however, the developer guide says 0-180 so we could assume degrees

Reply:
Please write this down in the documentation.